shared/src/api/action-adapter.coffee

```diff
@@ -139,19 +140,21 @@ connectModelAction = (action, apiHandler, klass, method, options) ->
       bind(this[options.onFail] , this, bind.placeholder, reqArgs, requestConfig)
     else
       apiHandler.getOptions().handlers.onFail
-    this.apiRequestsInProgress?.set(action, requestConfig)
+
+    this.api?.requestsInProgress.set(action, requestConfig)
     apiHandler.send(requestConfig, perRequestOptions, firstArg).then((reply) =>
-      this.apiRequestsInProgress?.delete(action)
+      this.api?.requestCounts[action] += 1
+      this.api?.requestsInProgress.delete(action)
       reply
-    ).catch( => this.apiRequestsInProgress?.delete(action))
+    ).catch( => this.apiRequestsInProgress.delete(action))
```

Comment:
seems like this should also be: `this.api?.requestsInProgress.delete(action)` when the XHR fails?

Reply:
Also, would of these changes need to be adapted for `tutor/src/models/map.js` as well?

Reply:
ah, yep.  I also missed some instances of `apiErrors`.  Fixing them now